src/main.rs

```diff
@@ -145,14 +144,73 @@ async fn main() -> Result<()> {
             )?;
 
             actors.push(actor);
-            error_futures.push(Box::pin(async move { error_ch.recv().await }));
+            error_channels.push(error_ch);
         }
     }
 
+    let (error_tx, mut error_rx) =
+        tokio::sync::mpsc::channel::<AntagonistError>(1);
+
+    for mut error_ch in error_channels {
+        let error_tx = error_tx.clone();
+        tokio::spawn(async move {
+            loop {
+                match error_ch.recv().await {
+                    Some(e) => {
+                        let _ = error_tx.send(e).await;
+                    }
+
+                    None => {
+                        let e = anyhow::anyhow!(
+                            "an antagonist disconnected its error channel!"
```

Comment:
It might be nice to be able to tell which one; since the antagonists know their names, we could add a `name` function to that trait and call it around line 155 to get an owned name string that we can move into the task.

Reply:
done in fc88d28